common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -1338,13 +1364,11 @@ def _flag_publish_event(self, course_key):
         Arguments:
             course_key - course_key to which the signal applies
         """
-        signal_handler = getattr(self, 'signal_handler', None)
-        if signal_handler:
-            bulk_record = self._get_bulk_ops_record(course_key) if isinstance(self, BulkOperationsMixin) else None
-            if bulk_record and bulk_record.active:
-                bulk_record.has_publish_item = True
-            else:
-                signal_handler.send("course_published", course_key=course_key)
+        bulk_record = self._get_bulk_ops_record(course_key) if isinstance(self, BulkOperationsMixin) else None
+        if bulk_record and bulk_record.active:
+            bulk_record.has_publish_item = True
+        else:
+            self.signal_handler.send("course_published", course_key=course_key)
```

Comment:
The new code is not equivalent to the old code. In the old code, if `signal_handler` is `None`, none of the code is executed. In the new version, the code is always executed. It might not make a difference, but wanted to point it out.